### PR TITLE
static expiration box

### PR DESCRIPTION
**Pull request by emma-green**
This should look like this: https://openedx.atlassian.net/wiki/spaces/RS/pages/2527068482/Expiration+Box

### CODE REVIEW

File: src/course-home/outline-tab/widgets/UpgradeCard.jsx [hunk 1]
```diff
@@ -1,21 +1,232 @@
...
 
-function UpgradeCard({ courseId, intl, onLearnMore }) {
-  const { org } = useModel('courseHomeMeta', courseId);
-  const {
-    offer,
-    verifiedMode,
-  } = useModel('outline', courseId);
+function UpsellNoFBECardContent() {
+  return (
+    <ul className="fa-ul upgradeCard-ul pt-0">
```
Comment (julianajlk) [inline src/course-home/outline-tab/widgets/UpgradeCard.jsx]: I would name my class as `upgrade-card-ul`, React doesn't have a restriction that I know of (dashed, camel-case, etc) but would be good for consistency with the other classNames already in this file

File: src/course-home/outline-tab/widgets/UpgradeCard.jsx [hunk 2]
```diff
@@ -1,21 +1,232 @@
...
+    );
+  }
+  return (<div className="upsell-warning">{expirationText}</div>);
+}
+
+ExpirationCountdown.propTypes = {
+  hoursToExpiration: PropTypes.number.isRequired,
+};
+
+function AccessExpirationDateBanner({ accessExpirationDate, timezoneFormatArgs }) {
```
Comment (julianajlk) [inline src/course-home/outline-tab/widgets/UpgradeCard.jsx:185]: What's the difference between a "card" and a "banner"? I think of them as the same, so to me when I read AccessExpirationDateBanner I think it's a separate banner/card than the UpgradeCard, but in this case it's a piece of it?

Comment (julianajlk) [reply src/course-home/outline-tab/widgets/UpgradeCard.jsx:185]: Also wondering if it would make sense to pull out some of this stuff to a separate file, and/or a messages.js file. It's hard to follow through all of the different functions, but if that would make it complicated to access the data we need here, nevermind

File: src/course-home/outline-tab/widgets/UpgradeCard.jsx [hunk 3]
```diff
@@ -1,21 +1,232 @@
 import React, { useEffect } from 'react';
 import PropTypes from 'prop-types';
+import { faCheck } from '@fortawesome/free-solid-svg-icons';
```
Comment (jhan217) [inline src/course-home/outline-tab/widgets/UpgradeCard.jsx:3]: Could you use the `check` icon from https://paragon-edx.netlify.app/foundations/icons?

File: src/course-home/outline-tab/widgets/UpgradeCard.jsx [hunk 4]
```diff
@@ -1,21 +1,232 @@
...
+  return (
+    <ul className="fa-ul upgradeCard-ul pt-0">
+      <li>
+        <span className="fa-li upgradeCard-li"><FontAwesomeIcon icon={faCheck} /></span>
+        <FormattedMessage
+          id="learning.outline.alert.upgradecard.verifiedCertLink"
+          defaultMessage="Earn a {verifiedCertLink} of completion to showcase on your resume"
+          values={{
+            verifiedCertLink: (
+              <a className="inline-link-underline font-weight-bold" rel="noopener noreferrer" target="_blank" href={`${getConfig().MARKETING_SITE_BASE_URL}/verified-certificate`}>verified certificate</a>
```
Comment (jhan217) [inline src/course-home/outline-tab/widgets/UpgradeCard.jsx:22]: Could you use a [Hyperlink](https://paragon-edx.netlify.app/components/hyperlink/) component here?

File: src/course-home/outline-tab/widgets/UpgradeCard.scss [hunk 5]
```diff
@@ -1,4 +1,57 @@
...
+  border-radius: 0 !important;
 }
+
+.upgrade-card-header{
+  margin: 1.25rem;
+
+}
+
+.upsell-warning{
+  background-color: #FCF1F4;
```
Comment (jhan217) [inline src/course-home/outline-tab/widgets/UpgradeCard.scss]: `$danger-100` from https://paragon-edx.netlify.app/foundations/colors?

File: src/course-home/outline-tab/widgets/UpgradeCard.scss [hunk 6]
```diff
@@ -1,4 +1,57 @@
...
+}
+
+.upsell-warning-light{
+  background-color: #FFFAED;
```
Comment (jhan217) [inline src/course-home/outline-tab/widgets/UpgradeCard.scss]: Should this be `$warning-100`?

File: src/course-home/outline-tab/widgets/UpgradeCard.scss [hunk 7]
```diff
@@ -1,4 +1,57 @@
...
+}
+
+.upsell-warning, .upsell-warning-light{
+  padding-left: 1.25rem;
```
Comment (jhan217) [inline src/course-home/outline-tab/widgets/UpgradeCard.scss:19]: Would it be possible to use any of the [spacing utility classes](https://paragon-edx.netlify.app/foundations/spacing) instead of hardcoding all the values?

File: src/course-home/outline-tab/widgets/UpgradeCard.scss [hunk 8]
```diff
@@ -1,4 +1,57 @@
...
+.discount-info {
+  border-top: 1px solid rgba(0, 0, 0, 0.125);
+}
+
+.inline-link-underline {
+  text-decoration: underline;
+}
+
+.upgrade-card .upgrade-card-message a{
+  color: #00262B;
```
Comment (jhan217) [inline src/course-home/outline-tab/widgets/UpgradeCard.scss]: `$primary-500`?

**Review by jhan217 — COMMENTED** [review]
Couple of general comments about potentially using Paragon variables/components

File: src/course-home/outline-tab/widgets/UpgradeCard.scss (hunk 7 above)
Comment (emma-green) [reply src/course-home/outline-tab/widgets/UpgradeCard.scss:19]: Good call out, but I don't think so, because I am not using full rems.

File: src/course-home/data/api.js [hunk 9]
```diff
@@ -206,6 +206,7 @@ export async function getOutlineTabData(courseId) {
     resumeCourse,
     verifiedMode,
     welcomeMessageHtml,
+    timeOffsetMillis: 0,
```
Comment (inventhouse) [inline src/course-home/data/api.js]: I don't see the code where this is actually computed from the `Date` headers (see these two diffs: [src/courseware/data/api.js](https://github.com/edx/frontend-app-learning/pull/385/files#diff-9a08a1a98535680aabde6f7b44130d363b29ea26469f026235f1f0a19e31a02f), [src/course-home/data/api.js](https://github.com/edx/frontend-app-learning/pull/385/files#diff-4d3a6d853ff9fd1354a56024f06c6718c856c281e295de92c5b09ac1e86774bf))

**Review by inventhouse — CHANGES_REQUESTED** [review]

File: src/course-home/data/api.js (hunk 9 above)
Comment (inventhouse) [reply src/course-home/data/api.js]: It would be good to include in the QA as well, that the countdown is correct no matter what your local computer time is set to.

Comment (emma-green) [reply src/course-home/data/api.js]: how about I write a test for it?

File: src/course-home/outline-tab/widgets/UpgradeCard.jsx (hunk 3 above)
Comment (emma-green) [reply src/course-home/outline-tab/widgets/UpgradeCard.jsx:3]: nope. It needs to be fontawesome in order to use it in a list

File: src/course-home/outline-tab/widgets/UpgradeCard.jsx (hunk 4 above)
Comment (emma-green) [reply src/course-home/outline-tab/widgets/UpgradeCard.jsx:22]: What would be the advantage of that?

Comment (jhan217) [reply src/course-home/outline-tab/widgets/UpgradeCard.jsx:22]: If the behavior/attributes of Hyperlinks changed, those changes would automatically apply to this link (if it were a Hyperlink component) as well. Not a big deal, as it looks like both are used in this repo.

File: src/course-home/outline-tab/widgets/UpgradeCard.jsx (hunk 2 above)
Comment (emma-green) [reply src/course-home/outline-tab/widgets/UpgradeCard.jsx:185]: There's a banner inside the card that counts down to expiration.

File: src/course-home/outline-tab/widgets/UpgradeCard.scss (hunk 5 above)
Comment (emma-green) [reply src/course-home/outline-tab/widgets/UpgradeCard.scss]: yes! Good catch! Thank you!!!

File: src/course-home/outline-tab/widgets/UpgradeCard.scss (hunk 6 above)
Comment (emma-green) [reply src/course-home/outline-tab/widgets/UpgradeCard.scss]: Hmmm....That's not the same color as requested. I'll message Eugene -- maybe FFFAED was an error

File: src/course-home/outline-tab/widgets/UpgradeCard.scss (hunk 8 above)
Comment (emma-green) [reply src/course-home/outline-tab/widgets/UpgradeCard.scss]: yep! Thanks!

File: src/course-home/outline-tab/widgets/UpgradeCard.scss (hunk 6 above)
Comment (emma-green) [reply src/course-home/outline-tab/widgets/UpgradeCard.scss]: She said to do $warning-100. Thanks for the callout!

File: src/course-home/data/api.js (hunk 9 above)
Comment (emma-green) [reply src/course-home/data/api.js]: I did a first pass. Please let me know your thoughts :)

**Review by inventhouse — APPROVED** [review]

File: src/course-home/outline-tab/widgets/UpgradeCard.scss [hunk 10]
```diff
@@ -1,4 +1,59 @@
-.outline-sidebar-upgrade-card {
-  border: 1px solid $dark-500;
-  border-top: 5px solid $dark-500;
+.upgrade-card {
```
Comment (mikix) [inline src/course-home/outline-tab/widgets/UpgradeCard.scss:1]: This file has a lot of custom css (which also means using a lot of `rem` instead of the bootstrap spacing system).

We prefer to do this sort of thing as bootstrap utility classnames like `m-2` or similar. Keeps sizes the same and reduces all the little bits of css lying around.

Even the stuff you can't do as bootstrap class (like `text-decoration`) we will often do inline to the jsx with a `style={{ textDecoration: 'underline' }}` bit, to keep the styling close to the component.

File: src/course-home/outline-tab/widgets/UpgradeCard.jsx (hunk 3 above)
Comment (mikix) [reply src/course-home/outline-tab/widgets/UpgradeCard.jsx:3]: Is that a bug with our paragon implementation? Seems like something we should put on a backlog or try to solve. We are trying to get away from direct uses of FA.

File: src/course-home/outline-tab/OutlineTab.jsx [hunk 11]
```diff
@@ -224,10 +225,14 @@ function OutlineTab({ intl }) {
...
+                  offer={offer}
+                  verifiedMode={verifiedMode}
+                  accessExpiration={accessExpiration}
+                  contentTypeGatingEnabled={datesBannerInfo.contentTypeGatingEnabled}
+                  userTimezone={userTimezone}
+                  timeOffsetMillis={timeOffsetMillis}
                   courseId={courseId}
-                  onLearnMore={
-                    canShowUpgradeSock ? () => { courseSock.current.showToUser(); } : null
-                  }
```
Comment (mikix) [inline src/course-home/outline-tab/OutlineTab.jsx:230]: Dropping this removes the need for the `ref` on `<CourseSock>`, so you can drop that bit and variable it saved the ref to.

File: src/course-home/data/__snapshots__/redux.test.js.snap [hunk 12]
```diff
@@ -432,6 +432,7 @@ Object {
           "hasVisitedCourse": false,
           "url": "http://localhost:18000/courses/course-v1:edX+DemoX+Demo_Course/jump_to/block-v1:edX+Test+Block@12345abcde",
         },
+        "timeOffsetMillis": 0,
```
Comment (mikix) [inline src/course-home/data/__snapshots__/redux.test.js.snap:435]: I don't see this being set in the upgrade card tests, and it's zero here. Is it worth a test where this is set and we see the change in the UI? (Or maybe there is one and I missed it)

**Review by mikix — COMMENTED** [review]
Some minor comments, nothing crazy / blocking
